### DISCUSSION TITLE
I can't install any mv macos version due to wrong vitual disks config

### DISCUSSION THREAD

**Discussion by Jeremi360** [quickemu-project/quickemu#1500] [Quickemu]
So I had working macos vm using quickemu in the past, but I removed it.
Now I need it again, but when I try to install using project [tutorial](https://github.com/quickemu-project/quickemu/wiki/03-Create-macOS-virtual-machines).
I tried both `macos big-sur` and `macos sonoma`, but in both cases it creates worng vm disks,
as it tutorial is said:
> Click **Disk Utility** and **Continue**
Select `QEMU HARDDISK Media` (~103.08GB) from the list (on Big Sur and above use `Apple Inc. VirtIO Block Device`) and click **Erase**.
But for me it creates those disks with inverted sizes:
- `Apple Inc. VirtIO Block Device` have (~103.08GB)
- `QEMU HARDDISK Media` have only few MB
![image](https://github.com/user-attachments/assets/bc8518d2-cfc7-4133-b5f3-be9f3d1c0e28)
![image](https://github.com/user-attachments/assets/cd37e0d9-5b89-46ac-a84a-d96845c41835)

So I tryied to ignore that and **Erase** and install it on `Apple Inc. VirtIO Block Device`,
but it dosen't work :(

**Comment by YuvanMichaelVivenzi** [comment]
Are you using the git version? I've installed macOS a few days ago with the stable one and it had no issue of disks. 

Git should be fine too.

# Try This: 
* Try the git version >> Run this in terminal ``` git clone https://github.com/quickemu-project/quickemu.git ```
* ``` cd ``` into the newly created quickemu directory and Run the command ``` ./quickget macos sonoma ```
* Let it do its thing, Later, after it's done, there should be a file called ``` macos-sonoma.conf ``` or something like that with the name of the OS you downloaded and ``` .conf ``` at the end of it. 
* Go into that file and make necessary changes. Here's what I have 
``` 
#!/usr/bin/quickemu --vm  # It'll point to the directory where you have the git version up.
guest_os="macos"   # should be the same
disk_img="macos-catalina/disk.qcow2"  # I have catalina, It should be almost the same for sonoma too.
img="macos-catalina/RecoveryImage.img"  # Respective sonoma image will be automatically made.
disk_size="128G"  # You can change disk size here
macos_release="catalina" # It'll say sonoma here
cpu_cores="4"  # Your wish, have as many cpu cores as you want
ram="16G"       # Same as with cpu cores, put a value to this as you wish
width="1360"   # Screen resolution configured in height and width, change it to your screen's measure
height="768"
```

* Run the vm with ``` ./quickemu --vm macos-sonoma.conf ``` 
* Go to disk utility
* Select the disk with the biggest size and format to ``` APFS ``` with the name ``` macOS ``` 
* Exit disk utility
* Go to ``` Reinstall macOS Sonoma ```
* Agree to the agreement and click on the disk named ``` macOS ``` to install.
* Let the install procedure complete, the vm should automatically reboot and get you into the first user setup.

**Reply by Jeremi360** [reply]
Thanks I will try this, I'm using AUR version.

**Reply by Jeremi360** [reply]
I checked version I'm using is 4.9.6 - I tryed AUR version again as I thought it was updated yesterday.
Sorry, but this doesn't work I tried using:
> Select the disk with the biggest size and format to APFS with the name macOS

And then Reinstall MacOS on this disk, but after 2 hours my vm is was just empty black screen:
![image](https://github.com/user-attachments/assets/464599de-b2fc-4069-b989-7eca3f0aabf1)
And after closing vm and resting it boot menu was the same :/

**Reply by Jeremi360** [reply]
Never mind, I no longer need it, so problem solved it self.

**Reply by ruben-benaco** [reply]
same problem. These steps fixed it for me:

> Select the disk with the biggest size and format to APFS with the name macOS
> Exit disk utility
> Go to Reinstall macOS Sonoma
> Agree to the agreement and click on the disk named macOS to install.

For some reason, the disk named QEMU HARDDISK Media has only exactly 402.7 MB and the instruction mention this disk specifically.
